dask/array/reductions.py

```diff
@@ -142,17 +142,17 @@ def reduction(x, chunk, aggregate, axis=None, keepdims=False, dtype=None,
     # Map chunk across all blocks
     inds = tuple(range(x.ndim))
     # The dtype of `tmp` doesn't actually matter, and may be incorrect.
-    tmp = blockwise(chunk, inds, x, inds, axis=axis, keepdims=True, dtype=x.dtype)
+    tmp = blockwise(chunk, inds, x, inds, axis=axis, keepdims=True, dtype=dtype or float)
```

Comment:
Why can't `x.dtype` be used anymore?

Reply:
Sometimes we construct Dask arrays that don't have array-like metadata.  This tends to happen in reductions a lot where we pass through tuples or dicts rather than arrays.  In these cases, calling `x.dtype` will err.

Reply:
Wouldn't it be better than to have `getattr(x, 'dtype', dtype)` instead?

Reply:
it doesn't really matter here.  The dtype of tmp isn't very relevant.  We ignore it in the following lines.  We just need to not call `x.dtype`, which can now sometimes err.

Reply:
Alright, thanks for clarifying.